app/controllers/v1/user_roles_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+module V1
+  class UserRolesController < BaseController
+    before_action :authenticate_request
+
+    def create
+      @ministry = Ministry.find_by(min_code: params[:ministry])
+      render_error_not_found and return if @ministry.nil?
+      add_admin
```

Comment:
Again, see PeopleController; it uses a nifty DRY pattern that comes from the "Growing Rails Applications" book.